code/modules/projectiles/guns/magnetic/bore.dm

```diff
@@ -111,6 +120,18 @@
 			user.visible_message("<span class='notice'>\The [user] slots \the [capacitor] into \the [src].</span>")
 			update_icon()
 			return
+		if(istype(thing, /obj/item/weapon/stock_parts/manipulator))
+			if(manipulator)
+				to_chat(user, "<span class='warning'>\The [src] already has \a [manipulator] installed.</span>")
+				return
+			manipulator = thing
+			user.drop_from_inventory(manipulator)
+			manipulator.forceMove(src)
+			playsound(loc, 'sound/machines/click.ogg', 10,1)
+			mat_cost = mat_cost % (2*manipulator.rating)
```

Comment:
The `mat_cost` used in the equation would need to be an `initial(mat_cost)` reference instead, as this currently means you could just take out the manipulator then re-install it to continually make it smaller.